metadata_layer.py

```diff
@@ -898,67 +941,73 @@ def apply(self, pool, replace=None):
-                if (pool.last_checked and (self.last_checked >= pool.last_checked)):
-                    # Update availabily information. This may result in the issuance
-                    # of additional events.
-                    changed_licenses = pool.update_availability(
-                        new_licenses_owned=self.licenses_owned,
-                        new_licenses_available=self.licenses_available,
-                        new_licenses_reserved=self.licenses_reserved,
-                        new_patrons_in_hold_queue=self.patrons_in_hold_queue,
-                        as_of=self.last_checked
-                    )
-
-        # Changes to the delivery mechanisms may have changed the work's
-        # open-access status.
-        old_open_access = pool.open_access                    
-        for lpdm in pool.delivery_mechanisms:
+        # Changes to the delivery mechanisms may have changed the
+        # open-access status of all LicensePools for this work.
+        pools = identifier.licensed_through
+        old_open_access = any(pool.open_access for pool in pools)
+        for lpdm in identifier.delivery_mechanisms:
             if (lpdm.rights_status
                 and lpdm.rights_status.uri in RightsStatus.OPEN_ACCESS):
-                pool.open_access = True
+                new_open_access = True
                 break
         else:
-            pool.open_access = False
-        open_access_status_changed = (old_open_access != pool.open_access)
+            new_open_access = False
+        if old_open_access != new_open_access:
+            for pool in pools:
+                pool.open_access = new_open_access
+        open_access_status_changed = (old_open_access != new_open_access)
+                    
+        # Finally, if we have data for a specific Collection's license
+        # for this book, find its LicensePool and update it.
+        changed_availability = False
+        if pool and self._availability_needs_update(pool):
+            # Update availabily information. This may result in
+            # the issuance of additional circulation events.
+            changed_availability = pool.update_availability(
+                new_licenses_owned=self.licenses_owned,
+                new_licenses_available=self.licenses_available,
+                new_licenses_reserved=self.licenses_reserved,
+                new_patrons_in_hold_queue=self.patrons_in_hold_queue,
+                as_of=self.last_checked
+            )
                     
-        made_changes = (made_changes or changed_licenses
+        made_changes = (made_changes or changed_availability
                         or open_access_status_changed)
 
         return pool, made_changes
 
-
-
+    def _availability_needs_update(self, pool):
+        """Does this CirculationData represent information more recent than 
+        what we have for the given LicensePool?
+        """
+        if not self.last_checked:
```

Comment:
You changed self.last_checked so that it's always defined. Does that break this check?